packages/server/src/Document.ts

```diff
@@ -121,11 +124,25 @@ export class Document extends Doc {
     return this
   }
 
+  addDirectConnection(): Document {
```

Comment:
do we need this (`directConnectionsCount`) actually? 🤔

Reply:
The scenario I'm guarding against here (I think) is:
- a direct connection is opened
- a regular connection is opened, modifications made, then closed
- the usual "clean this document out of memory" code (https://github.com/ueberdosis/hocuspocus/blob/main/packages/server/src/Hocuspocus.ts#L785) should not be triggered, because the direct connection still has a claim on the document

Testing this at https://github.com/ueberdosis/hocuspocus/pull/610/files#diff-f23cea543b7ce0921f130f63eae2ec74a5941c3486239317a010ee8a6b5970b8R4